src/signals.c

```diff
@@ -324,9 +327,8 @@ static void sig_handle_timer(int signum)
 #endif
 #ifndef BENCH_BUILD
 #if OS_TIMER
-	/* Some stuff only done every few seconds */
-	if (timer_save_interval < 4 ||
-	    ((timer_save_interval - timer_save_value) & 3) == 3) {
+	/* Some stuff only done every third second */
+	if (((timer_save_interval - timer_save_value) & 3) == 1) {
```

Comment:
More like every fourth.

I guess the check for `== 1` instead of `== 3` is meant to trigger at least once per interval even if the interval is very short, thus eliminating the need for checking it for `< 4`? Have you tested that this is actually triggered for those short intervals?

Reply:
Good call. Without testing yet, I guess it should be `== 0` but I will review this closer.